src/nsrt_learning/strips_learning/gen_to_spec_learner.py

```diff
@@ -229,23 +254,19 @@ def _backchain_one_pass(
                     # matched to another PNAD.
                     self._recompute_datastores_from_segments(
                         param_opt_to_nec_pnads[option.parent])
+                    # Now that we have done this, certain PNADs may be
+                    # left with empty datastores. Remove these.
+                    self._remove_empty_datastore_pnads(param_opt_to_nec_pnads,
```

Comment:
check whether there's any other places where you could make use of this new function -- i think we filter out empty datastore pnads in a couple of places?

Reply:
there's one other place (line 106 in this file), but using our function here seems like it'll be much uglier than the current code.